src/hydrusvideodeduplicator/dedup.py

```diff
@@ -237,6 +238,29 @@ def clear_search_cache() -> None:
         except OSError:
             rprint(f"[red] Database does not exist. Cannot clear search cache.")
 
+    # Updates the 'farthest_search_index' of all videos in the database that have an index greater than the length of
+    # the database, reducing them down to the length of the database.
+    @staticmethod
+    def update_search_cache(new_total: int | None = None) -> None:
+
+        if new_total is not None and new_total < 1:
+            raise Exception(f"passed in total must be greater than 0, but received {new_total}")
+        try:
+            CHUNK_SIZE = 32
+            with SqliteDict(str(DEDUP_DATABASE_FILE), tablename="videos", flag="c", outer_stack=False) as hashdb:
+                if new_total is None:
+                    new_total = len(hashdb)
+                for batched_keys in HydrusVideoDeduplicator.batched(hashdb, CHUNK_SIZE):
+                    for key in batched_keys:
+                        row = hashdb[key]
+                        if 'farthest_search_index' in row and row['farthest_search_index'] > new_total:
+                            row['farthest_search_index'] = new_total
+                            hashdb[key] = row
+                    hashdb.commit()
+        except Exception as exc:
+            rprint(f"[red] Error encountered when updating search indices:\n{str(exc)}")
+            rprint("[red] Warning: duplicate search for this run may not be fully complete.")
```

Comment:
I'm removing the messages for interrupting the process since they don't matter. It will run on next startup anyway.